packages/common/src/ConfiguFile.ts

```diff
@@ -65,35 +85,47 @@ export const ConfiguFileSchema: JsonSchemaType<ConfiguFileContents> = {
         required: ['type'],
         properties: {
           type: { type: 'string' },
-          configuration: { type: 'object', nullable: true },
-          backup: { type: 'boolean', nullable: true },
+          configuration: { type: 'object' },
+          backup: { type: 'boolean' },
         },
       },
-      nullable: true,
+      // nullable: true,
     },
-    backup: ConfiguFileSchemaDefs.StringProperty,
-    schemas: ConfiguFileSchemaDefs.StringMapProperty,
-    scripts: ConfiguFileSchemaDefs.StringMapProperty,
+    backup: StringPropertySchema,
+    schemas: StringMapPropertySchema,
+    scripts: StringMapPropertySchema,
+    // todo: add ticket to support register api
   },
-};
+} as const satisfies JSONSchemaObject;
+
+export type ConfiguFileContents = FromSchema<typeof ConfiguFileSchema>;
 
 export class ConfiguFile {
+  public static readonly flags = ConfiguFileSchemaId;
+  public static readonly envs = ['CONFIGU_CONFIG', 'CONFIGU_CONFIG_CONFIGURATIONS'];
+  // public static readonly cacheDirName = '/cache';
+  public static readonly schema = ConfiguFileSchema;
   constructor(
     public readonly path: string,
     public readonly contents: ConfiguFileContents,
   ) {
-    if (!JsonSchema.validate({ schema: ConfiguFileSchema, path, data: this.contents })) {
-      throw new Error(`ConfiguFile.contents "${path}" is invalid\n${JsonSchema.getLastValidationError()}`);
+    try {
+      JSONSchema.validate(ConfiguFile.schema, this.contents);
+    } catch (error) {
+      throw new Error(`ConfiguFile.contents "${path}" is invalid\n${error.message}`);
     }
   }
 
   private static async init(path: string, contents: string): Promise<ConfiguFile> {
-    // expend contents with env vars
-    // todo: find a way to escape template inside Expression class
-    const { value: renderedContents, error } = Expression.parse(`\`${contents}\``).tryEvaluate(process.env);
-    if (error || typeof renderedContents !== 'string') {
+    // try expend contents with env vars
+    let renderedContents: string;
+    try {
+      // todo: find a way to escape template inside Expression class
```

Comment:
```suggestion
```
This comment is irrelevant, the `evaluateTemplateString` method does the escaping.